cli/tests/test_config_command.rs

```diff
@@ -125,6 +125,72 @@ fn test_config_list_all() {
     "###);
 }
 
+#[test]
+fn test_config_list_show_sources() {
+    let mut test_env = TestEnvironment::default();
+    test_env.jj_cmd_ok(test_env.env_root(), &["init", "repo", "--git"]);
+    let repo_path = test_env.env_root().join("repo");
+
+    // Create multiple user configs.
+    test_env.add_config(
+        r#"
+        user-key-1 = "user-val-1"
+        "#,
+    );
+
+    test_env.add_config(
+        r#"
+        user-key-2 = "user-val-2"
+        "#,
+    );
+
+    // Env
+    test_env.add_env_var("env-key", "env-value");
+
+    // Repo
+    test_env.jj_cmd_ok(
+        &repo_path,
+        &["config", "set", "--repo", "repo-key", "repo-val"],
+    );
+
+    let stdout = test_env.jj_cmd_success(
+        &repo_path,
+        &[
+            "config",
+            "list",
+            "--config-toml",
+            "cmd-key='cmd-val'",
+            "--show-sources",
+        ],
+    );
+
+    // Paths starting with `$TEST_ENV` confirm that the relative path returned by
+    // `Value.origin()` has been converted to an absolute path.
+    insta::assert_snapshot!(stdout, @r###"
+    usercfg:$TEST_ENV/config/config0001.toml	template-aliases.format_time_range(time_range)="time_range.start() ++ \" - \" ++ time_range.end()"
+    usercfg:$TEST_ENV/config/config0002.toml	user-key-1="user-val-1"
+    usercfg:$TEST_ENV/config/config0003.toml	user-key-2="user-val-2"
+    repocfg:$TEST_ENV/repo/.jj/repo/config.toml	repo-key="repo-val"
```

Comment:
Not sure if the colon placement will make it easier to parse - there is already a tab there that parsers can split on.

In Git, looking at the commit that introduced this feature (70bd879), I don't think there was an explicit decision to unify all the files - it was just about exposing what the parsing mechanism already knew. Support for `--show-scope` (to differentiate between the different types of files) was added later.

Reply:
> Not sure if the colon placement will make it easier to parse - there is already a tab there that parsers can split on.

True. I think it still helps humans parse it if the colon consistently comes after the source/origin.

Reply:
I've switched to the proposed syntax. To my eye it's a hard to tell where the path ends and the value starts on longer lines.

```
env: ui.pager="/usr/bin/less"
usercfg /home/user/.config/jj/config.toml: ui.editor="nvim"
usercfg /home/user/.config/jj/config.toml: user.email="wfc@wfchandler.org"
usercfg /home/user/.config/jj/config.toml: user.name="Will Chandler"
repocfg /home/user/devel/jj/.jj/repo/config.toml: colors.commit_id prefix.bold=true
repocfg /home/user/devel/jj/.jj/repo/config.toml: revset-aliases.immutable_heads()="main@origin | gh-pages@origin"
cmdline: ui.paginate="never"
```

Also changed `command line` to `cmdline` for easier splitting.

Reply:
> To my eye it's a hard to tell where the path ends and the value starts on longer lines.

Oh, we should use the formatter to label the different parts of the output so it can be styled differently, but that's out of scope for this PR.

Reply:
Makes sense 👍